crates/apollo-encoder/src/operation.rs

```diff
@@ -0,0 +1,257 @@
+use std::fmt;
+
+use crate::{Directive, SelectionSet, VariableDef};
+
+/// The __operationDef type represents an operation definition
+///
+/// *OperationDefinition*:
+///     OperationType Name? VariableDefinitions? Directives? SelectionSet
+///
+/// Detailed documentation can be found in [GraphQL spec](https://spec.graphql.org/October2021/#sec-Language.Operations).
+///
+/// ### Example
+/// ```rust
+/// use apollo_encoder::{Argument, Field, InlineFragment, Directive, OperationDef, OperationType, Selection, SelectionSet, TypeCondition, Type_, Value, VariableDef};
+/// use indoc::indoc;
+///
+/// let selection_set = {
+///     let sels = vec![
+///         Selection::Field(Field::new(String::from("first"))),
+///         Selection::Field(Field::new(String::from("second"))),
+///     ];
+///     let mut sel_set = SelectionSet::new();
+///     sels.into_iter().for_each(|sel| sel_set.selection(sel));
+///
+///     sel_set
+/// };
+/// let var_def = VariableDef::new(
+///     String::from("variable_def"),
+///     Type_::List {
+///         ty: Box::new(Type_::NamedType {
+///             name: String::from("Int"),
+///         }),
+///     },
+/// );
+/// let mut new_op = OperationDef::new(OperationType::Query, selection_set);
+/// let mut directive = Directive::new(String::from("testDirective"));
+/// directive.arg(Argument::new(
+///     String::from("first"),
+///     Value::String("one".to_string()),
+/// ));
+/// new_op.variable_definition(var_def);
+/// new_op.directive(directive);
+///
+/// assert_eq!(
+///     new_op.to_string(),
+///     indoc! { r#"
+///         query($variable_def: [Int]) @testDirective(first: "one") {
+///           first
+///           second
+///         }
+///     "#}
+/// );
+/// ```
+#[derive(Debug)]
+pub struct OperationDef {
+    operation_type: OperationType,
+    name: Option<String>,
+    variable_definitions: Vec<VariableDef>,
+    directives: Vec<Directive>,
+    selection_set: SelectionSet,
+}
+
+impl OperationDef {
+    /// Create a new instance of OperationDef
+    pub fn new(operation_type: OperationType, selection_set: SelectionSet) -> Self {
+        Self {
+            operation_type,
```

Comment:
How do you handle query shorthand encoding that doesn't have the "query" keyword, but is indeed of `Query` operation type?

Reply:
It was not supported. It's fixed :)